gender_novels/analysis/analysis.py

```diff
@@ -285,6 +290,161 @@ def instance_dist(novel, word):
+                print("ERR: no MALE values: " + authlst[num])
+            if shelst == 0:
+                print("ERR: no FEMALE values: " + authlst[num])
+    return d
+
+
+def bubble_sort_across_lists(dictionary):
+    """
+    :param dictionary: containing 3 different list values
+    :return dictionary sorted across all three lists in a specific method:
+    """
+    lst1 = dictionary['he']
+    lst2 = dictionary['she']
+    lst3 = dictionary['book']
+    r = range(len(lst1) - 1)
+    p = True
+
+    # sort by lst1 descending
+    for j in r:
+        for i in r:
+            if lst1[i] < lst1[i + 1]:
+                # manipulating lst 1
+                temp1 = lst1[i]
+                lst1[i] = lst1[i + 1]
+                lst1[i + 1] = temp1
+                # manipulating lst 2
+                temp2 = lst2[i]
+                lst2[i] = lst2[i + 1]
+                lst2[i + 1] = temp2
+                # manipulating lst of authors
+                temp3 = lst3[i]
+                lst3[i] = lst3[i + 1]
+                lst3[i + 1] = temp3
+                p = False
+        if p:
+            break
+        else:
+            p = True
+
+    # sort by lst2 ascending
+    for j in r:
+        for i in r:
+            if lst2[i] > lst2[i + 1]:
+                # manipulating lst 1
+                temp1 = lst1[i]
+                lst1[i] = lst1[i + 1]
+                lst1[i + 1] = temp1
+                # manipulating lst 2
+                temp2 = lst2[i]
+                lst2[i] = lst2[i + 1]
+                lst2[i + 1] = temp2
+                # manipulating lst of authors
+                temp3 = lst3[i]
+                lst3[i] = lst3[i + 1]
+                lst3[i + 1] = temp3
+                p = False
+        if p:
+            break
+        else:
+            p = True
+    d = {}
+    d['he'] = lst1
+    d['she'] = lst2
+    d['book'] = lst3
+    return d
+
+
+def instance_stats(book, medians1, medians2, title):
+    fig, ax = plt.subplots()
+    plt.ylim(0, 1000)
+
+    index = np.arange(len(book))
+    bar_width = .7
+
+    medians_she = tuple(medians2)
+    medians_he = tuple(medians1)
+    book = tuple(book)
+
+    rects1 = ax.bar(index, medians_he, bar_width, color='cyan', label='Male to Female')
+
+    rects2 = ax.bar(index, medians_she, bar_width, color='plum', label='Female to Male')
+
+    ax.set_xlabel('Book')
+    ax.set_ylabel('Ratio of Median Values')
+    ax.set_title(
+        'MtF or FtM Ratio of Median Distance of Gendered Instances by Author')
+    ax.set_xticks(index)
+    plt.xticks(fontsize=8, rotation=90)
+    ax.set_xticklabels(book)
+    ax.set_yscale("symlog")
+
+    ax.legend()
+
+    fig.tight_layout()
+    # plt.show()
+    filepng = "visualizations/" + title + ".png"
+    filepdf = "visualizations/" + title + ".pdf"
+    plt.savefig(filepng, bbox_inches='tight')
+    plt.savefig(filepdf, bbox_inches='tight')
+
+
+def run_dist_inst():
+    corpus = Corpus('sample_novels')
```

Comment:
this needs at least documentation, if not doctests.  Lab policy is: write documentation, then code.

Why is 10 hardcoded?  What does it mean?